src/main/java/data/Database.java

```diff
@@ -0,0 +1,69 @@
+package data;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+
+/* This class creates an instance of a database that a file can be written to and read from.
+Assumptions: for each runID, only 1 of each datatype can be associated with it */
+public class Database {
+  /* Place where files are stored.*/
+  private String name;
+
+  /* List of all files stored in Database.*/
+  ArrayList<String> files;
+
+  /* Creates instance of a database with this name. */
+  public Database(String name) {
+    /* Database will be GCS and buckets. Bucket will named by different runids. */
+    this.name = name;
+  }
+
+  /* Returns projectID of Database. */
+  public String getName() {
+    return name;
+  }
+
+  // Attempt to open a stream to the data, normally to database, in this case to
+  // file.
+  public InputStream readData(String fileName) {
```

Comment:
Document when a function's return parameter (or argument) may be null. Otherwise, we generally assume arguments and returns are non-nullable, because it's easier to reason about.

https://drive.google.com/file/d/1It0D63aBJZE_NGqMHgBTEZbtF0Y79WYI/view?usp=sharing